src/dom/index.ts

```diff
@@ -21,40 +21,40 @@ export function render(code: () => any, element: Node): () => void {
   return disposer!;
 }
 
-export function For<T, U>(props: {
+export function For<T, U extends JSX.Element>(props: {
   each: T[];
-  fallback?: any;
+  fallback?: U;
```

Comment:
Forcing the fallback to be the same type as children seems like a bad idea. Have you checked that this actually works? Can fallback be a single element or string while children returns multiple elements?

Reply:
I was thinking about this and I think I'm going to remove children. Element is a possible multi-type too. I realized the types there were redundant since Element already handles arrays and functions.